zanata-war/src/main/java/org/zanata/events/DocStatsEvent.java

```diff
@@ -0,0 +1,58 @@
+package org.zanata.events;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.zanata.common.ContentState;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+import javax.annotation.Nullable;
+
+/**
+ * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
+ */
+@AllArgsConstructor
+@Getter
+@EqualsAndHashCode
+public class DocStatsEvent {
+    private final DocumentLocaleKey key;
+
+    private final Long projectVersionId;
+
+    /**
+     * Updated content states with word counts
+     */
+    private final Map<ContentState, Integer> contentStates;
+
+    private final Long lastModifiedTargetId;
+
+    public static Map<ContentState, Integer> updateContentState(
+        @Nullable Map<ContentState, Integer> contentStates, ContentState newState,
+        ContentState previousState, Long longWordCount) {
```

Comment:
If we're going to accept Long here, we should have a Map to Long. Given that we've chosen to use Long for HTextFlow.wordCount, we should probably be consistent. Do we need to accept nullable Longs, or could we change it to long?